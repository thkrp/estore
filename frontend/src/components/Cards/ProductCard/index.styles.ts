import styled from 'styled-components';
import Link from '../../Links/Link';
import { theme } from '../../../styles/theme';
import { media } from '../../../styles/media';

export const CardStyled = styled(Link)`
    flex: 0 0 25%;
    padding: 12px;
    border: 1px solid ${theme.colors.borders.lightgrey};
    text-decoration: none;
    display: flex;
    flex-flow: column;
    height: 100%;
    background: ${theme.colors.main.white};
    position: relative;

    &:hover {
        z-index: 2;
        border-color: ${theme.colors.borders.darkgrey};
        box-shadow: none;
    }

    &.slider {
        margin: 15px 0;
    }

    ${media.untilTabletLg`
        flex: 0 0 33.333333333%;
    `}
    ${media.untilTabletLg`
        flex: 0 0 50%;
    `}
    ${media.untilPhoneMd`
        flex: 1 1 100%;
    `}
`;

export const ImageStyled = styled.span<{ $image: string }>`
    display: block;
    background: url('${({ $image }) => $image}') no-repeat center 10% / contain;
    width: 100%;
    flex: 1 1 30%;
    min-height: 200px;
`;

export const HeadingStyled = styled.h3`
    height: 80px;
    font-family: ${theme.fonts.bold};
    font-size: 22px;
    color: ${theme.colors.main.secondary};
    overflow: hidden;
`;
