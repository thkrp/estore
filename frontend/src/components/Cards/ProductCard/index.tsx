import React, { FC } from 'react';
import { Product } from 'app-shared';
import { imageUrl } from '../../../common/helpers/path.helper';
import { CardStyled, HeadingStyled, ImageStyled } from './index.styles';

interface Props {
    item?: Product;
    className?: string;
}

const ProductCard: FC<Props> = ({ item, className = '' }) => {
    if (!item) {
        return null;
    }

    return (
        <CardStyled to={item.url || '/'} key={item.code} className={className} role="link">
            <ImageStyled $image={imageUrl(item.image)} />
            <HeadingStyled title={item.name}>{item.name}</HeadingStyled>
        </CardStyled>
    );
};

export default ProductCard;
