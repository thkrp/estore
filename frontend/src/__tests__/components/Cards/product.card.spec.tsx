/* eslint-disable no-restricted-imports*/
import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { mockData, Product } from 'app-shared';
import * as Sanitizer from '@braintree/sanitize-url';
import ProductCard from '../../../components/Cards/ProductCard';

describe('Product card component', () => {
    let product: Product;
    beforeAll(() => {
        [product] = mockData.products.items;
    });

    it('should return null', () => {
        const { container } = render(<ProductCard />);
        expect(container).toBeEmptyDOMElement();
    });

    it('should return a card', () => {
        const { container } = render(<ProductCard item={product} />, { wrapper: MemoryRouter });
        expect(container).toMatchSnapshot();
    });

    it('should have a sanitized link', () => {
        const sanitizeUrl = jest.spyOn(Sanitizer, 'sanitizeUrl');
        render(<ProductCard item={product} />, { wrapper: MemoryRouter });
        expect(sanitizeUrl).toBeCalled();
    });

    it('should be able to pass a class', () => {
        render(<ProductCard item={product} className="brand" />, { wrapper: MemoryRouter });
        const element = screen.getByRole('link');
        const classes = element.className.split(' ');
        expect(classes.includes('brand')).toBe(true);
    });
});
