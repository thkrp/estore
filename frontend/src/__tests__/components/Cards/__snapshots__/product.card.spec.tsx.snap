// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card component should return a card 1`] = `
<div>
  <a
    class="sc-bcPKhP kwXmRT"
    href="/catalog/motoblochnoe-navesnoe-oborudovanie/adaptery-dlya-motoblokov/adapter-k-motobloku-ara"
    role="link"
  >
    <span
      class="sc-grXZZQ hQYoSY"
    />
    <h3
      class="sc-dkjKgF gnGgQT"
      title="Адаптер к мотоблоку \\"АРА\\""
    >
      Адаптер к мотоблоку "АРА"
    </h3>
  </a>
</div>
`;
